python/ray/serve/_private/router.py

```diff
@@ -296,18 +300,22 @@ def __init__(
         self,
         event_loop: asyncio.AbstractEventLoop,
         deployment_name: str,
+        self_node_id: Optional[str] = None,
     ):
         self._loop = event_loop
         self._deployment_name = deployment_name
+        self._self_node_id = self_node_id
 
         # Current replicas available to be scheduled.
         # Updated via `update_replicas`.
         self._replica_id_set: Set[str] = set()
         self._replicas: Dict[str, ReplicaWrapper] = {}
         self._replicas_updated_event = make_asyncio_event_version_compat(event_loop)
+        self._replica_ids_colocated_on_same_node: Set[str] = set()
```

Comment:
nit: Can we shorten this to just `_colocated_replica_ids` and call out in a comment that colocation assumes same node